File: pset6/mario.py
import cs50

while True:
    # prompt user for height
    print("Height: ", end = "")
    height = cs50.get_int()
    if height > 0 and height <= 23:
        break

for row in range(1, height + 1):
    # print spaces
    print(" " * (height - row), end = "")
    # print hashes
    print("#" * row, end = "")
    # print gap (2 spaces)
    print("  ", end = "")
    #print hashes
    print("#" * row)

File: pset1/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    // prompt the user for the half-pyramids' height, a non-negative integer no greater than 23 and validate
    int height;
    do
    {
        printf("Height: ");
        height = get_int();
    }
    while (height < 0 || height > 23);

    // generate the half-pyramids
    for (int i = 0; i < height; i++)
    {
        // print spaces for left pyramid
        for (int j = 1 ; j < height - i; j++)
        {
            printf(" ");
        }
        // print hashes for left pyramid
        for (int j = 0; j <= i; j++ )
        {
            printf("#");
        }
        // print gap
        printf("  ");
        // print hashes for right pyramid
        for (int j = 0; j <= i; j++)
        {
            printf("#");
        }
        // print new line
        printf("\n");
    }

}